controller/controller_server/utils.py

```diff
@@ -14,36 +16,140 @@
 logger = get_stdout_logger()
 
 
-def get_array_connection_info_from_secret(secrets):
-    user = secrets[config.SECRET_USERNAME_PARAMETER]
-    password = secrets[config.SECRET_PASSWORD_PARAMETER]
-    array_addresses = secrets[config.SECRET_ARRAY_PARAMETER].split(",")
+def _is_topology_match(secret_topologies, node_topologies):
+    is_match = False
+    for topologies in secret_topologies:
+        logger.debug(
+            "Comparing topologies: object topologies: {},"
+            " node topologies: {}".format(topologies, node_topologies))
+        if all(item in node_topologies.items() for item in topologies.items()):
```

Comment:
We compare only if we have an item in secret_topologies which is ⊆ node_topologies. We don't compare the other way: node_topologies ⊆ secret_topologies.
Does it bother us? If no, Why?

Reply:
It's supposed to be this way.
`node_topologies` contains all of the worker topology keys the define it. the `secret_topologies` contain the bare minimum to identify the system by it, so if several workers have the same topologies or a common one, it's enough to only mention them to identify the system. the other way is not working cause we don't need to identify the worker..